onnx/test/inference_function_test.py

```diff
@@ -196,6 +197,85 @@ def test_scan_inference_with_subgraph(self) -> None:
             }
         )
 
+    def test_inference_with_conflow(self) -> None:
+        input = """
+        <
+            ir_version: 8,
+            opset_import: ["" : 18, "onnxscript.atenlib" : 1],
+            producer_name: "pytorch",
+            producer_version: "2.1.0"
+        >
+        torch_jit (float input_0) => (float reault, int64 index)
+        {
+            reault, index = onnxscript.atenlib.aten_min_dim <dim = 0, keepdim = 1> (input_0)
+        }
+        <
+            domain: "onnxscript.atenlib",
+            opset_import: ["" : 18]
+        >
+        aten_min_dim <dim>(self) => (result_7, indices_6)
+        {
+            tmp = Shape (self)
+            tmp_0 = Size (tmp)
+            tmp_1 = Constant <value = int64 tmp_1 {0}> ()
+            tmp_1_cast = CastLike (tmp_1, tmp_0)
+            tmp_2 = Equal (tmp_0, tmp_1_cast)
+            cond = Not (tmp_2)
+            indices_6, result_7 = If (cond) <
+                then_branch = thenGraph_4 () => ( indices,  result) {
+                    dim = Constant <value_int: int = @dim> ()
+                    tmp_3 = Constant <value_ints = [-1]> ()
+                    dims = Reshape (dim, tmp_3)
+                    result = ReduceMin <keepdims: int = @keepdim> (self, dims)
+                    indices = ArgMin <axis: int = @dim, keepdims: int = @keepdim> (self)
+                }, else_branch = elseGraph_4 () => ( indices_4,  result_5) {
+                    indices_4 = Constant <value_int = 0> ()
+                    result_5 = Identity (self)
+                }
+            >
+        }
+        """
+        model = onnx.parser.parse_model(input)
+        onnx.shape_inference.infer_shapes(model, strict_mode=False)
+        with self.assertRaises(onnx.shape_inference.InferenceError):
+            onnx.shape_inference.infer_shapes(model, strict_mode=True)
+
+    def test_inference_with_attribute(self) -> None:
```

Comment:
Out of curiosity: what is being tested here? Just wondering how it is connected to this PR.

Reply:
Without the fix in this PR, this test will fail because strict_mode=True will not be enforced for the 'then' subgraph.